### PR TITLE
clusterversion: set `binaryMinSupportedVersion` to `v21_1`

### LINKED ISSUE #67277: clusterversion: set `binaryMinSupportedVersion` on master to `v21_1`
**Is your feature request related to a problem? Please describe.**

We should probably have done this when we introduced `Start21_2`. Internal discussion [here](https://cockroachlabs.slack.com/archives/C4A9ALLRL/p1624272123033500) and [here](https://cockroachlabs.slack.com/archives/C0HM2DZ34/p1625589314499100). Does not depend on but does relate to #33578.

### PR DESCRIPTION
**Pull request by ajwerner**
This probably should have been done when `Start21_2` was added.

Fixes #67277.

Release note: None

### CODE REVIEW

**Review by knz — APPROVED** [review]

**Comment by rafiss** [comment]
Just stumbled into this; some process Qs:
1. do we need to prioritize this specific issue within one of the engineering teams?
2. do we need to add something to a process somewhere so some team (i guess release-eng?) does this for future releases?

**Comment by knz** [comment]
does this need help?

**Comment by ajwerner** [comment]
This has been rebased and most of the failures removed because of https://github.com/cockroachdb/cockroach/pull/67822. 

I added another commit to fix a real problem. I am left somewhat confused: why are we running the migration before bumping to the fence version? @irfansharif am I missing something or is this migration call in the wrong place?

**Comment by ajwerner** [comment]
>  @irfansharif am I missing something or is this migration call in the wrong place?

I think I've got it sorted. We do it to make sure the version gating for new nodes joining is sound

**Comment by ajwerner** [comment]
I would like very much to backport the 2nd commit. Who needs to look at it and what additional testing does it need?

**Review by knz — COMMENTED** [review]
1st, 3rd and 4th commit LGTM

Would like a review from @irfansharif on the 2nd commit too

> Reviewed 5 of 5 files at r4, 2 of 3 files at r5, 2 of 2 files at r7, 1 of 1 files at r8.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67281#-:-Mg6x2hXEgvf8C8fMQRO:b-rgbonp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner)

___
*[pkg/migration/migrationmanager/manager.go, line 295 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/67281#-Mg6uJvrBcmYEtpvSNGs:-Mg6uJvrBcmYEtpvSNGt:b-5kpo56) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb5f13a1cc2e0b01336fe99e7d52f40298ba0f2c/pkg/migration/migrationmanager/manager.go#L295)):*
> ```Go
> 		alreadyCompleted, err = migrationjob.CheckIfMigrationCompleted(ctx, txn, m.ie, version)
> 		if alreadyCompleted || err != nil {
> 			log.Infof(ctx, "already completed or error %v", err)
> ```

maybe split the cases and use `log.Errorf` where appropriate?

**Review by ajwerner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/67281)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz)

___
*[pkg/migration/migrationmanager/manager.go, line 295 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/67281#-Mg6uJvrBcmYEtpvSNGs:-MgB7HDy3oiOcAVs-yh7:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb5f13a1cc2e0b01336fe99e7d52f40298ba0f2c/pkg/migration/migrationmanager/manager.go#L295)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

maybe split the cases and use `log.Errorf` where appropriate?
</blockquote></details>

Done.

**Review by irfansharif — APPROVED** [review]
LGTM.

> why are we running the migration before bumping to the fence version? @irfansharif am I missing something or is this migration call in the wrong place?

> We do it to make sure the version gating for new nodes joining is sound

Yup. The other thing to note is that the release demarcation versions (`Start21_2`, `Start21_1`, etc.) don't have any migrations attached to them. The migration manager will then:
```go
runMigration(..., Start21_2)
bumpClusterVersion(..., fenceVersionFor(Start21_2))
runMigration(..., <first real version in 21.2>)
```

As I say this I realize we don't have panics/asserts making sure we don't accidentally attach a migration on one of these special `StartXY_Z` versions. We could document/assert in this PR if you feel it necessary.

**Comment by irfansharif** [comment]
(Also sorry for the review lag, only now crawling out of my inbox after a hectic L2 rotation + a long weekend.)

File: pkg/migration/migrations/truncated_state_external_test.go [hunk 1]
```diff
@@ -39,21 +44,36 @@ func TestTruncatedStateMigration(t *testing.T) {
...
-				// When the active cluster version is greater than it, we assert
-				// against the presence of legacy truncated state and ensure
-				// we're using the range applied state key. In this test we'll
-				// start of at the version immediately preceding the migration,
-				// and migrate past it.
-				BinaryVersionOverride: clusterversion.ByKey(clusterversion.TruncatedAndRangeAppliedStateMigration - 1),
-				// We want to exercise manual control over the upgrade process.
-				DisableAutomaticVersionUpgrade: 1,
+			makeArgs := func() (args base.TestServerArgs) {
+				minVersion := minVersion
```
[the remark this answers is not in the document]
Comment (ajwerner) [reply pkg/migration/migrations/truncated_state_external_test.go]: Detritus. I think at some point for some reason there were some pointers involved. Removed.

**Comment by ajwerner** [comment]
TFTR!

bors r+

**Pull request merged**